validators/cgroup_validator_linux.go

```diff
@@ -44,28 +44,117 @@ func (c *CgroupsValidator) Name() string {
 }
 
 const (
-	cgroupsConfigPrefix = "CGROUPS_"
-	unifiedMountpoint   = "/sys/fs/cgroup"
+	cgroupsConfigPrefix      = "CGROUPS_"
+	defaultUnifiedMountpoint = "/sys/fs/cgroup"
+	mountsFilePath           = "/proc/mounts"
+	mountInfoPath            = "/proc/self/mountinfo"
 )
 
+// getUnifiedMountpoint checks if the default mount point is available.
+// If not, it parses the mounts file to find a valid cgroup mount point.
+func getUnifiedMountpoint() (string, error, error) {
+	isCgroupfs, err := isMountPointCgroupfs(defaultUnifiedMountpoint)
+	if isCgroupfs {
+		return defaultUnifiedMountpoint, err, nil
+	}
+	mountPath, err := parseMountsFile(mountsFilePath)
+	return mountPath, err, fmt.Errorf("%q is not cgroupfs: %v", defaultUnifiedMountpoint, err)
+}
+
+func isMountPointCgroupfs(path string) (bool, error) {
```

Comment:
i think this should be unit tested.

you can break it down into two functions:
isMountPointCgroupfs opens that file and reads the []byte / string with isMountPointCgroupfsImpl()

impl == implementation. we it this way in some places in kubeadm.
isMountPointCgroupfsImpl should be unit tested.